lightning/src/chain/mod.rs

```diff
@@ -332,22 +332,14 @@ pub trait Filter {
 	fn register_tx(&self, txid: &Txid, script_pubkey: &Script);
 
 	/// Registers interest in spends of a transaction output.
-	///
-	/// Optionally, when `output.block_hash` is set, should return any transaction spending the
-	/// output that is found in the corresponding block along with its index.
-	///
-	/// This return value is useful for Electrum clients in order to supply in-block descendant
-	/// transactions which otherwise were not included. This is not necessary for other clients if
-	/// such descendant transactions were already included (e.g., when a BIP 157 client provides the
-	/// full block).
-	fn register_output(&self, output: WatchedOutput) -> Option<(usize, Transaction)>;
```

Comment:
Could we update the commit message with the reasoning for why we're removing the return value? And maybe why we're removing that test too?

Reply:
Done!